home/default.nix

```diff
@@ -3,33 +3,46 @@ with lib;
 
 let
 
-  cfg = config.colorschemes.kauz;
+  cfg = config.kauz;
 
-in {
+  supported = [ "fish" "kitty" "neovim" "tmux" ];
 
-  options.colorschemes.kauz = {
-    enable = mkEnableOption "a colorscheme for neovim, kitty, fish and tmux";
-  };
+  capitalizeFirst = s:
+    lib.toUpper (builtins.substring 0 1 s)
+    + builtins.substring 1 (lib.stringLength s) s;
 
-  config = mkIf cfg.enable {
+  isFish = cfg.fish.enable;
+  isKitty = cfg.kitty.enable;
+  isTmux = cfg.tmux.enable;
+  isNeovim = cfg.neovim.enable;
 
-    programs.kitty.extraConfig = ''
-      include ${pkgs.kauz-kitty}/kauz.conf
-    '';
+in {
 
-    programs.tmux.extraConfig = builtins.readFile "${pkgs.kauz-tmux}/kauz.tmux";
+  options.kauz = builtins.listToAttrs (map (s: {
```

Comment:
I think I would unroll the options, too. It will allow us to have more fine grained options for certain apps, in case that's ever needed.